pkg/services/ngalert/notifier/channels/slack.go

```diff
@@ -0,0 +1,244 @@
+	if mentionChannel != "" && mentionChannel != "here" && mentionChannel != "channel" {
+		return nil, alerting.ValidationError{
+			Reason: fmt.Sprintf("Invalid value for mentionChannel: %q", mentionChannel),
+		}
+	}
+
+	mentionUsersStr := model.Settings.Get("mentionUsers").MustString()
+	mentionUsers := []string{}
+	for _, u := range strings.Split(mentionUsersStr, ",") {
+		u = strings.TrimSpace(u)
+		if u != "" {
+			mentionUsers = append(mentionUsers, u)
+		}
+	}
+
+	mentionGroupsStr := model.Settings.Get("mentionGroups").MustString()
+	mentionGroups := []string{}
+	for _, g := range strings.Split(mentionGroupsStr, ",") {
+		g = strings.TrimSpace(g)
+		if g != "" {
+			mentionGroups = append(mentionGroups, g)
+		}
+	}
+
+	return &SlackNotifier{
+		NotifierBase:   old_notifiers.NewNotifierBase(model),
+		URL:            slackURL,
+		Recipient:      recipient,
+		MentionUsers:   mentionUsers,
+		MentionGroups:  mentionGroups,
+		MentionChannel: mentionChannel,
+		Username:       model.Settings.Get("username").MustString("Grafana"),
+		IconEmoji:      model.Settings.Get("icon_emoji").MustString(),
+		IconURL:        model.Settings.Get("icon_url").MustString(),
+		Token:          model.DecryptedValue("token", model.Settings.Get("token").MustString()),
+		Text:           model.Settings.Get("text").MustString(`{{ template "slack.default.text" . }}`),
+		Title:          model.Settings.Get("title").MustString(`{{ template "slack.default.title" . }}`),
+		Fallback:       model.Settings.Get("fallback").MustString(`{{ template "slack.default.title" . }}`),
+		AutoResolve:    model.Settings.Get("autoResolve").MustBool(true),
+		log:            log.New("alerting.notifier.slack"),
+		tmpl:           t,
+		externalUrl:    externalUrl,
+	}, nil
+}
+
+// slackMessage is the slackMessage for sending a slack notification.
+type slackMessage struct {
+	Channel     string                   `json:"channel,omitempty"`
+	Username    string                   `json:"username,omitempty"`
+	IconEmoji   string                   `json:"icon_emoji,omitempty"`
+	IconURL     string                   `json:"icon_url,omitempty"`
+	Attachments []attachment             `json:"attachments"`
+	Blocks      []map[string]interface{} `json:"blocks"`
+}
+
+// attachment is used to display a richly-formatted message block.
+type attachment struct {
+	Title      string              `json:"title,omitempty"`
+	TitleLink  string              `json:"title_link,omitempty"`
+	Text       string              `json:"text"`
+	Fallback   string              `json:"fallback"`
+	Fields     []config.SlackField `json:"fields,omitempty"`
+	Footer     string              `json:"footer"`
+	FooterIcon string              `json:"footer_icon"`
+	Color      string              `json:"color,omitempty"`
+	Ts         int64               `json:"ts,omitempty"`
+}
+
+func getAlertStatusColor(status model.AlertStatus) string {
+	if status == model.AlertFiring {
+		return "#D63232"
```

Comment:
Should these be configurable?

Reply:
Currently we used fixed colors, so I think it need not be configurable even now

Reply:
Let's move them to a constant then 😄